### ISSUE TITLE
Dragging an event in a modal on top of another calendar

### ISSUE THREAD

**Issue by numsu** [fullcalendar/fullcalendar#5026]
We have an implementation which uses the fullcalendar's week/month view on bottom and when clicking an event, it opens up specifics of that event in a modal with a smaller day view calendar to display the day's other events etc. We have an implementation where we can drag the event in the modal window in order to change the start/end time.

Changing the end time works OK, but when dragging the whole event to change the start time, it gets confused by the full calendar which is under the modal window. The event which is being dragged appears on the far left side of the calendar which is below the modal and cannot be dragged on the modal calendar.

We use it with the angular plugin but it's reproducible in the plain version too.

Codepen: https://codepen.io/numsu/pen/NWKXONq

![image](https://user-images.githubusercontent.com/6938280/64504637-43476780-d2d9-11e9-979b-d61b51760934.png)

**Comment by acerix** [comment]
Thanks for the report and demo.  I can see the same issue with a bootstrap modal:

https://codepen.io/acerix/pen/zYOWBBa

**Comment by ricardo-lobo** [comment]
@acerix Do know any workaround for this? 

I'm having the same issue, and the only way I managed to "solve" this, was to hide (display: none) the background calendar element.

**Comment by numsu** [comment]
Has there been any progress on this issue? Previously I didn't mention that it worked well before the V4 update. If there's something in the old implementation which could help to solve the issue.

Also the issue is not only about being unable to drag the event on the calendar on top, but also when dragging, the event being dragged is appearing to float in a different position than where the drag started. It can be seen in the image and my initial pen. This isn't happening on your pens @acerix.

Hiding the calendar below feels like a bad solution and it will confuse users, I'm hoping for this to be resolved soon-ish :)

**Comment by OFMaurice** [comment]
For me i did the following workaround:

- added function 'hackForFullcalendarIssue5026' to your code
- call it on the modal calendar once after it's rendered

see https://codepen.io/OFMaurice/pen/NWPPMqG

**Comment by gongAll** [comment]
@OFMaurice that workaround works for the normal calendar over calendar scenario. Won't work for external events being dragged within the popup. Do you have any idea of how to achieve this? Have been trying for a couple of hours with no success.

**Comment by gongAll** [comment]
So I've been able to understand why I can't apply the workaround to the external draggable elements. I believe this small change would allow me to do so:

![DeepinScreenshot_20200515143018](https://user-images.githubusercontent.com/36202212/82055658-bd29bf00-96b8-11ea-8605-23878acbffb5.png)

It's just a matter of having access to the HitDragging element correspondent to the ExternalDraggable object.

@acerix @arshaw do you think this could be merged to v4 as a quick hotfix? I really need this to pass it all to production.

**Comment by LaurenceMommers** [comment]
I have the same issue, however it sounds like someone already tought about it in the code.
The Hit class returned from the queryHit function should return a layer property which is checked in HitDragging.queryHitForOffset to find the best hit.

However all implementations of queryHit i found return 0 for the layer e.g.  https://github.com/fullcalendar/fullcalendar/blob/f55ba69a8247dda0cc6bb0c4aefceed21bf17eb9/packages/daygrid/src/DayTable.tsx#L98

Is it possible to verify that my assumtion is correct and this property is indeed ment to resemble something like a z-Index?

- After some searching i don't think it is possible to get the actual rendering order in javascript (note, zIndex itself returns an empty string).

another solution might be to use document.elementsFromPoint(10, 10) and then filter the mirror object. The challenge in this would be that it changes the design a bit and it returns the element but now you have to find a way to get the correct droppableStore.

**Comment by Taaqif** [comment]
Also facing this issue, only option we found is to not render the calendar when the modal is open or to hide the calendar body using display: none; A third option is to navigate the user to a different screen, but that breaks the user flow in certain scenarios.

Both options aren't ideal as in both scenarios, the user sees a glitch happen behind the calendar which has been reported a few times. Depending on calendar contents, the page can also jump around also causing the user to think there is an error happening in the background.  

Really keen on finding a solution to this issue.

**Comment by ganger97** [comment]
Has there been any progress on this issue? Its still bug more than 2 year. I need this feature to my react application. Do know any workaround for this without hide of original calendar? Will be fix that in some future update?

**Comment by faraby7** [comment]
@numsu Yes I agree with you hiding the calendar will confuse the user the only way at the moment is that before I open the second Fullcalendar I change the view for the first calendar to ListMonth.

fullCalendarRef.current.getApi().changeView("listMonth");

**Comment by jomskie0123** [comment]
Any progress on this issue?

**Comment by snezhnyikorol** [comment]
Any update on this issue? 4.5 years since issue created and no change despite the fact many users reported it and it is really usual use case but nobody cares

**Comment by arshaw** [comment]
very sorry about the lack of progress here. I'll take a look at this for the next bugfix release.

**Comment by arshaw** [comment]
The root cause of this bug is now fixed in [v6.1.12](https://github.com/fullcalendar/fullcalendar/releases/tag/v6.1.12), thought with some caveats.

Updated repro: https://codepen.io/arshaw/pen/rNgeRZL?editors=0110

Caveats: The `transform: translateY(#)` CSS statements is confusing FullCalendar's coordinate system. When removed, things work fine. I've commented out those lines in the demo. I need to investigate further as CSS-translations are increasingly common.

**Comment by arshaw** [comment]
The remaining positioning bug of the mirror element is due to this:
https://achrafkassioui.com/blog/position-fixed-and-CSS-transforms/

It's because the mirror-element is position:fixed and within the calendar, which gets messed up when there's a transform.

The temporary solution is do change the `fixedMirrorParent` to `document.body`
https://fullcalendar.io/docs/fixedMirrorParent

For @numsu's updated recreation, I also had to ensure the modal's z-index was below the z-index that FullCalenadar uses for the drag-mirror elements, which is 9999.

So, with the correct precautions, this bug should be considered fixed. Please respond if there are any continued issues.

**Comment by kaylen-keynote9** [comment]
@numsu https://typescripter.space/questions/unusual-interactions-between-angular-and-x3d-technologies
